[MRG+1] Remove the MLComp text categorization example

rth opened the pull request:
This PR fixes issue https://github.com/scikit-learn/scikit-learn/issues/8229 (document classification example should use 'latin-1' encoding) by removing the example (as [suggested](https://github.com/scikit-learn/scikit-learn/issues/8229#issuecomment-276587605) by @lesteve ).


This also raises a deprecation warning when `load_mlcomp` is used to load the 20 newsgoups example, where `fetch_20newsgroups` should preferably be used instead. However, as both datasets are [not strictly identical](https://github.com/scikit-learn/scikit-learn/issues/8229#issuecomment-276421104), maybe that's not the best solution.

The [MLComp text categorization example](http://scikit-learn.org/stable/auto_examples/text/mlcomp_sparse_document_classification.html) is mostly redundant with the [other text categorization example](http://scikit-learn.org/stable/auto_examples/text/document_classification_20newsgroups.html) while being less complete, and incites user to use a more complex way of loading the 20 newsgoups dataset via `load_mlcomp` instead of  `fetch_20newsgroups`.

rth committed:
Removing the mlcomp_sparse_document_classification.py

rth committed:
Add deprecation warning

lesteve commented:
> This also raises a deprecation warning when load_mlcomp is used to load the 20 newsgoups example, where fetch_20newsgroups should preferably be used instead. However, as both datasets are not strictly identical, maybe that's not the best solution.

Chatting with @ogrisel he agrees that deprecating `load_mlcomp`. According to http://mlcomp.org/, they are going to shut down the website on March 2017.

lesteve commented on sklearn/datasets/mlcomp.py:
```diff
@@ -68,6 +68,11 @@ def load_mlcomp(name_or_id, set_="raw", mlcomp_root=None, **kwargs):
         raise ValueError("Could not find folder: " + mlcomp_root)
 
+    if name_or_id in ['20news-18828', '20news-19997', '20news-bydate']:
+        raise DeprecationWarning("please consider using "
```
Deprecate the whole function with a `@deprecated` decorator. Also mention which version it will be deprecated in (0.19) and which version it will be removed (0.21). Look either at the contributing guidelines or other deprecation messages in the scikit-learn code.

rth committed:
Addressing review comments

rth commented:
Thanks for the review @lesteve . I addressed your comments.

lesteve commented:
> The MLComp text categorization example is mostly redundant with the other text categorization example while being less complete

Can I get a second opinion on this?

jnothman reviewed (commented):
Perhaps note somewhere in docs (deprecation message or docstring) that the site itself is closing down. Otherwise LGTM

rth committed:
Mentioned the that the website is closing in the warning message

rth commented:
Thanks for the review @jnothman ! I addressed your comment.
CI is failing due to https://github.com/travis-ci/travis-ci/issues/7264, will try to trigger a new build a bit later..

lesteve commented:
The Travis problem has been fixed, I have restarted the build.

lesteve commented:
Looks good, merging, thanks a lot!
